plugins/package-managers/pub/src/main/kotlin/Pubspec.kt

```diff
@@ -0,0 +1,153 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+package org.ossreviewtoolkit.plugins.packagemanagers.pub
+
+import com.charleskorn.kaml.Yaml
+import com.charleskorn.kaml.YamlConfiguration
+import com.charleskorn.kaml.YamlInput
+import com.charleskorn.kaml.YamlMap
+import com.charleskorn.kaml.YamlNode
+import com.charleskorn.kaml.YamlScalar
+import com.charleskorn.kaml.yamlMap
+import com.charleskorn.kaml.yamlScalar
+
+import java.io.File
+
+import kotlinx.serialization.KSerializer
+import kotlinx.serialization.SerialName
+import kotlinx.serialization.Serializable
+import kotlinx.serialization.SerializationException
+import kotlinx.serialization.decodeFromString
+import kotlinx.serialization.descriptors.serialDescriptor
+import kotlinx.serialization.encoding.Decoder
+import kotlinx.serialization.encoding.Encoder
+
+import org.ossreviewtoolkit.plugins.packagemanagers.pub.Pubspec.Dependency
+import org.ossreviewtoolkit.plugins.packagemanagers.pub.Pubspec.GitDependency
+import org.ossreviewtoolkit.plugins.packagemanagers.pub.Pubspec.HostedDependency
+import org.ossreviewtoolkit.plugins.packagemanagers.pub.Pubspec.PathDependency
+import org.ossreviewtoolkit.plugins.packagemanagers.pub.Pubspec.SdkDependency
+
+private val YAML = Yaml(configuration = YamlConfiguration(strictMode = false))
+
+internal fun parsePubspec(pubspecFile: File): Pubspec = parsePubspec(pubspecFile.readText())
+
+internal fun parsePubspec(pubspecYaml: String): Pubspec = YAML.decodeFromString(pubspecYaml)
+
+/**
+ * See https://dart.dev/tools/pub/pubspec.
+ */
+@Serializable
+internal data class Pubspec(
+    val name: String,
+    val version: String? = null,
+    val description: String? = null,
+    val homepage: String? = null,
+    val author: String? = null,
+    val authors: Set<String> = emptySet(),
+    val repository: String? = null,
+    val sdk: String? = null,
+    @Serializable(DependencyMapSerializer::class)
+    val dependencies: Map<String, Dependency>? = null,
+    @Serializable(DependencyMapSerializer::class)
+    @SerialName("dev_dependencies")
+    val devDependencies: Map<String, Dependency>? = null
+) {
+    @Serializable
+    sealed interface Dependency
+
+    @Serializable
+    data class HostedDependency(
+        val version: String,
+        val url: String? = null
+    ) : Dependency
+
+    @Serializable
+    data class PathDependency(
+        val path: String
+    ) : Dependency
+
+    @Serializable
+    data class SdkDependency(
+        val sdk: String
+    ) : Dependency
+
+    @Serializable
+    data class GitDependency(
+        val url: String,
+        val path: String? = null,
+        val ref: String? = null
+    ) : Dependency
+}
+
+private class DependencyMapSerializer : KSerializer<Map<String, Dependency>> {
```

Comment:
Can this inherit from `StdDeserializer` instead to not have the "dummy override" for `serialize` (and to avoid the strict need for the `descriptor` override)?

Reply:
No, because `StdSerializer` extends from `JsonSerializer`, while this one is for `YAML`.

Reply:
Sorry, I mixed up the Jackson's `StdSerializer` with KxS's `JsonTransformingSerializer` anyway.

So I guess the missing KAML feature is https://github.com/charleskorn/kaml/issues/29? If so, how about adding serializer class docs with a reference to that issue?